### PR TITLE
Enable goa API endpoints as a fallback for the new API endpoints

### PR DESCRIPTION
**Pull request by rtnpro**
This change does the following things:
- Runs the new API endpoints at ``/api/``
- Run the goa API endpoints at ``/legacyapi/``
- Redirects from ``/api/*`` endpoint if route is missing to ``/legacyapi/``

This will enable a smooth migration from the Goa based API to the Gin based API, without any change in the client side.

### CODE REVIEW

File: api/gin_engine_setup_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"github.com/fabric8-services/fabric8-wit/api"
+	"github.com/fabric8-services/fabric8-wit/gormapplication"
+	"github.com/fabric8-services/fabric8-wit/gormsupport/cleaner"
+	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
+)
+
+// Execute submits the request and returns the response recording fo subseauent verifications
+func Execute(s gormtestsupport.GinkgoTestSuite, r *http.Request) *httptest.ResponseRecorder {
```
Comment (xcoulon) [inline api/gin_engine_setup_test.go]: we should avoid duplicating this utility method. What about moving this test in the `api/handler` pkg instead ? you could rename the `gin_engine_setup_test.go` file into `redirect_test.go` and verify that the call to `GET /api/foobar` is redirected, as you've already done in this PR. Also, could you verify that `POST`, `PATCH` and `DELETE` requests are also correctly handled ?

**Review by xcoulon — CHANGES_REQUESTED** [review]

File: api/gin_engine_setup_test.go (hunk 1 above)
Comment (rtnpro) [reply api/gin_engine_setup_test.go]: Yes

Comment (rtnpro) [reply api/gin_engine_setup_test.go]: Fixed!

**Comment by rtnpro** [comment]
@xcoulon I have updated the API redirection test cases for ``POST``, ``PATCH`` and ``DELETE`` request methods along with ``GET``.  Also, I have moved the common ``Execute`` function to ``api/test/utils.go`` so that it can be shared.

**Comment by xcoulon** [comment]
great work, @rtnpro ! The code looks good overall, so I'll merge it. Thanks !

**Pull request merged**